package.json

```diff
@@ -123,9 +123,10 @@
   "scripts": {
     "vscode:prepublish": "pnpm compile",
     "release": "bumpp package.json --commit --push --tag && git update-ref refs/heads/release refs/heads/main && git push origin release",
-    "compile": "tsup ./src/extension.ts --external vscode --minify",
-    "watch": "tsup ./src/extension.ts --external vscode --watch --sourcemap",
-    "test": "vitest run --config vite.global-config.ts",
+    "compile": "tsup ./src/extension.ts test/**.test.ts --external vscode --minify",
```

Comment:
Ah I don’t think vitest exposes a similar api, but maybe we can have the same script, but run it with “tsx script” instead of “node script” so it supports ts?

Reply:
Vscode's integration test is not so modern and it seems tricky to use typescript directly.

What I ended doing was to  create a wrapper to do `require("tsx/cjs")`  then re-export a custom "test runner script" https://github.com/hi-ogawa/vscode-extension-shell-shortcut/blob/main/src/test/run-wrapper.js

Reply:
Can't we just make it easy to start `pnpm watch:test`? That would make everything behave nicely. The compilation time is almost instant.

`vscode-test` also has a watch mode. We could utilize that too.

And if we don't like building tests, perhaps it'd feel better if we had a separate command that would only build the tests, and another command to only build the source?

Reply:
I don't necessary oppose transpiling vscode test files, but if there's a way to just plug-in `tsx` somewhere while still using `vscode-test` cli, then that would be great.

I haven't checked `vscode-test` cli, so let me dig into it a bit.

Reply:
I think we can use `mocha.preload` option to setup just `require("tsx/cjs")`. I tested it here https://github.com/hi-ogawa/vscode-extension-shell-shortcut/pull/11

Reply:
Great! Whenever I have time, I'll take a look at that! 😍

Reply:
That seemed to work! Awesome ❤️